connector-agent/src/data_sources/csv.rs

```diff
@@ -26,21 +35,56 @@ impl SourceBuilder for CSVSourceBuilder {
         }
     }
 
-    fn build(&mut self) -> Self::DataSource {
-        CSVSource::new()
+    fn set_queries<Q: AsRef<str>>(&mut self, queries: &[Q]) {
+        self.files = queries
+            .into_iter()
+            .map(|fname| fname.as_ref().to_string())
+            .collect();
+    }
+
+    fn fetch_metadata(&mut self) -> Result<()> {
+        let mut reader = csv::ReaderBuilder::new()
+            .has_headers(true)
```

Comment:
We assume csv file has header here, but assume it does not when do prepare. Should we let the user input whether the csv file contain the header and let the source have this information?

Reply:
let's first assume they all have headers. Once we want to make csv source production-ready we can add more parameters.